### ISSUE TITLE
rop.syscall attribute fails to find syscall gadget

### ISSUE THREAD

**Issue by nbulischeck** [Gallopsled/pwntools#1326]
This line fails to find valid syscall gadgets. I'm not 100% sure if that for loop is a valid solution.

https://github.com/Gallopsled/pwntools/blob/c15afc592a94a5fd4c1255d2ce0137be38164a66/pwnlib/rop/rop.py#L1276

```Python
>>> binary = ELF.from_assembly('syscall; ret')
>>> rop = ROP(binary)
>>> rop.syscall # No result
>>> [rop.gadgets[gadget] for gadget in rop.gadgets if 'syscall' in rop.gadgets[gadget]['insns']]
[Gadget(0x10000000, [u'syscall', u'ret'], [], 0x4)] # Syscall gadget
```

**Comment by saullocarvalho** [comment]
How can I help to fix this issue?
While reproducing the steps provided by @nbulischeck , I got the following:

```
>>> elf = ELF.from_assembly('syscall; ret')
>>> rop = ROP(elf)
>>> rop.syscall
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hunter/app/pwntools/pwnlib/rop/rop.py", line 1398, in __getattr__
    if self.gadgets[each]['insns'] == [mapping[attr]]:
  File "/home/hunter/app/pwntools/pwnlib/rop/gadgets.py", line 57, in __getitem__
    if isinstance(key, six.integer_types):
NameError: name 'six' is not defined
>>> rop.gadgets.items()
dict_items([(268435458, Gadget(0x10000002, ['ret'], [], 0x4)), (268435456, Gadget(0x10000000, ['syscall'], [], 0x0))])
```

`pwnlib/rop/gadgets.py` is missing `import six`. This is easy to fix.

Regarding the original issue, should the `syscall` gadget have only the `syscall` instruction or should `insns` also include `ret`?

**Comment by saullocarvalho** [comment]
`ROPgadget` is returning the `syscall` gadget and the gadget does not follow the instructions until it reaches a `ret`. That seems odd to me.

**Comment by Arusekk** [comment]
Fixed in #1678 and #1682.
